trestlebot/entrypoints/templates/github/trestlebot-rules-transform.yml

```
@@ -0,0 +1,45 @@
+---
+name: Trestle-bot rules-transform and autosync
+
+on:
+  pull_request:
```

```suggestion
  push:
```
I think we could use `dry_run` on pull request based workflow and `push` by default. The goal here to reduce merge conflict and review of generated content while still being able to verify it before merge. WDYT? This we would be the same change for all the sync-based templates.